rasa/core/domain.py

```diff
@@ -118,9 +129,21 @@ def from_dict(cls, data: Dict) -> "Domain":
             utter_templates,
             data.get("actions", []),
             data.get("forms", []),
+            session_config=session_config,
             **additional_arguments,
         )
 
+    @staticmethod
+    def _get_session_config(additional_arguments: Dict) -> SessionConfig:
+        session_length = additional_arguments.pop(
+            SESSION_LENGTH_KEY, DEFAULT_SESSION_LENGTH
+        )
+        carry_over_slots = additional_arguments.pop(
+            CARRY_OVER_SLOTS_KEY, DEFAULT_CARRY_OVER_SLOTS_TO_NEW_SESSION
+        )
+
```

Comment:
should we add a check that 

1. `session_length` must be a positive float
2. `carry_over_slots` must be of type bool

if either of those fail, we take default values and log a warning?

Reply:
I think it's enough if we validate through pykwalify, don't you think?

Reply:
yep, all good, didn't see you were already applying those in the schema 👍